clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -229,7 +229,11 @@ private GroupRebalanceConfig buildRebalanceConfig(Optional<String> groupInstance
     @AfterEach
     public void teardown() {
         this.metrics.close();
-        this.coordinator.close(time.timer(0));
+        try {
+            this.coordinator.close(time.timer(0));
```

Comment:
Close will throw now if our `asyncCommit` gets fenced, because we actually attempt to run the callback. If we don't want that, I don't think we can implement the guarantee that the callback is run in close.

Reply:
I see, I would say it's fine to throw the error at the coordinator level (and live with code like this). 

And actually, the need for this catch is not introduced by this PR as I see it. The coordinator close before this PR could throw fenced exception for async commits that were waiting for coord and completed [here](https://github.com/apache/kafka/blob/fd9c7d2932dee055289b403e37a0bbb631c080a9/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L983) getting fenced.

Reply:
correct. it was just less likely before.